daemon/cmd/daemon_main.go

```diff
@@ -1517,14 +1517,6 @@ func initEnv(cmd *cobra.Command) {
 		}
 	}
 
-	// This is necessary because the code inside pkg/k8s.NewService() for
-	// parsing services would not trigger unless NodePort is enabled. Without
-	// NodePort enabled, the external and LB IPs would not be parsed out.
-	if option.Config.BGPAnnounceLBIP {
```

Comment:
I'm confused by the following comment:

> previous to this commit, when converting from a slim_core Service to our
> k8s Service abstraction the parsed out loadBalancersIP were not
> converted to the appropriate map of net.IP objects.
> 
> they were not converted because of a guard which only does this for when
> "NodePort" configuration is set to true.

Existing code (i.e., before this patch), seems to enable `NodePort` whenever `BGPAnnounceLBIP` was set. So doesn't this mean that the IPs would always be parsed correctly?

Reply:
In thoery yes, in practice, other code flip that bool on and off, so its becomes a crap shoot on whether EnableNodePort *stays* true, based on other configuration. This is what caused the issue to begin with.